targets/TARGET_STM/gpio_irq_api.c

```diff
@@ -149,9 +149,13 @@ static void handle_interrupt_in(uint32_t irq_index, uint32_t max_num_pin_line)
 #else /* TARGET_STM32L5 */
 
             // Clear interrupt flag
+#if defined(DUAL_CORE) && defined(CORE_CM4)
+            if (__HAL_GPIO_EXTID2_GET_FLAG(pin) != RESET) {
+                __HAL_GPIO_EXTID2_CLEAR_FLAG(pin);
+#else
```

Comment:
A bit more information and code to reproduce the problem can be found here https://github.com/arduino/ArduinoCore-mbed/pull/68#issue-499769014 .

Reply:
Yes it looks good to distinguish between interrupts for Core1 and for Core2.
But I wonder if it a good chose to use CORE_CM4, because it is right that on H7  CM4 = Core2, but not sure it will be valid for further targets

Reply:
Let's keep CORE_CM4